pandas/_libs/src/parser/io.c

```diff
@@ -62,6 +66,11 @@ void *new_file_source(char *fname, size_t buffer_size) {
 #endif
     if (fs->fd == -1) {
         free(fs);
+#ifdef USE_WIN_UTF16
```

Comment:
Is this not applicable to the above calls? If it is you can just make a macro for it in `portable.h`

Reply:
I removed the special case handling for Windows. I checked the python code and PyErr_SetFromErrnoWithFilename works also in Windows